frontend/src/store/pinia-modules/oauth.ts

```diff
@@ -13,10 +13,57 @@ const convert = (raw: ResourceObject): OAuthToken => {
 
 export const useOAuthStore = defineStore("oauth", {
   actions: {
-    async exchangeVCSToken({ vcsId, code }: { vcsId: VCSId; code: string }) {
+    // exchangeVCSTokenWithID uses "vcsId" to let the backend infer the details from an existing VCS provider.
+    async exchangeVCSTokenWithID({
+      vcsId,
+      code,
+    }: {
+      vcsId: VCSId;
+      code: string;
+    }): Promise<OAuthToken> {
       const data = (
-        await axios.post(`/api/oauth/vcs/${vcsId}/exchange-token`, null, {
-          headers: { code },
+        await axios.post(`/api/oauth/vcs/exchange-token`, {
+          data: {
+            type: "exchangeToken",
+            attributes: {
+              code,
+              vcsId,
+            },
+          },
+        })
+      ).data.data;
+
+      const token = convert(data);
+      return token;
+    },
+
+    // exchangeVCSTokenWith passes client details to the backend to allow the backend directly
+    // compose the request to the VCS host. This should only be used in the initial VCS set up.
+    async exchangeVCSTokenWith({
```

Comment:
nit: exchangeVCSToken ?

Reply:
Addressed in https://github.com/bytebase/bytebase/pull/1007/commits/f132ce0cc48e3d56c547ffb18b496427bfdec909.